src/k8s/pkg/client/kubernetes/certificate_signing_request.go

```diff
@@ -0,0 +1,46 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+
+	certificatesv1 "k8s.io/api/certificates/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// WatchCertificateSigningRequest watches a CertificateSigningRequest with the given name and calls the verify function on each event.
+func (c *Client) WatchCertificateSigningRequest(ctx context.Context, name string, verify func(csr *certificatesv1.CertificateSigningRequest) (bool, error)) error {
```

Comment:
Make the semantics of how CSRs are treated part of the godoc of this function. Make it clear what we expect out of `verify`, make it clear _when_ this function terminates (e.g. it terminates when the CSR is approved or failed, etc).

The thing to avoid here is: in the future, someone wants to check a CSR, sees a `kubernetes.WatchCertificateSigningRequest`, but the semantics of the function are not clear.

This is important, as the name `WatchCertificateSigningRequest` does not communicate that the watch actually stops when the verify function returns `true, nil`. That should be documented clearly. Perhaps, the function itself should return `(bool, error)` as well, or maybe the final CSR object.

Reply:
In general, avoid "implicit" API contracts, be very explicit with what the function expects and returns.